runtime/lua/vim/lsp/util.lua

```diff
@@ -325,7 +325,13 @@ function M.jump_to_location(location)
   local bufnr = vim.uri_to_bufnr(location.uri)
   -- Save position in jumplist
   vim.cmd "normal! m'"
-  -- TODO(ashkan) use tagfunc here to update tagstack.
+
+  -- Push a new item into tagstack
+  local items = {}
+  table.insert(items, {tagname=vim.fn.expand("<cword>"), from=vim.fn.getpos('.')})
+  vim.fn.settagstack(vim.fn.bufnr('%'), {items=items}, 't')
```

Comment:
I'm not super familiar with the function, but why would you use `t` here as the argument. I thought `a` would be better for this case (since we're just adding a new thing to the stack).

Also, can you add a test for this? For example, call this and then check that the tag stack is updated in the tests?

Reply:
agreed. @yen3 have you been using this for awhile and it's working as expected? Can you try with 'a'?  Then we can merge.

Adding a test would be nice but shouldn't block this.

Reply:
I prefer to use `t` rather then `a` since it's possible to pop the tagstack then push again.

For example, `jump to` means `Ctrl-]` and `jump back to` means `Ctrl-t`

| Num | Action              | Stack                 | Current Index |
| --- | ------------------- | --------------------- | ------------- |
| 1   | jump to A1          |  A1                   | 1             |
| 2   | jump to B1          |  A1 -> B1             | 2             |
| 3   | jump to C1          |  A1 -> B1 -> C1       | 3             |
| 4   | jump to D1          |  A1 -> B1 -> C1 -> D1 | 4             |
| 5   | jump back to C1     |  A1 -> B1 -> C1 -> D1 | 3             |
| 6   | jump back to B1     |  A1 -> B1 -> C1 -> D1 | 2             |
| 7   | jump back to A1     |  A1 -> B1 -> C1 -> D1 | 1             |
| 8   | jump to B2          |  A1 -> B2             | 2             |

In num-7 to num-8 action, the stack content has to be manipulated by the current index then append new stack item.

If we use `a` to append in num-8, the stack is `A1 -> B1 -> C1 -> D1 -> B2` not `A1 -> B2`. The stack behaviour is not the same as original.

I have no idea how to add test case(s) for the project and run them. Is there any documentation to help to write test cases? If no, I will study it.

Thanks for your comment and discussion. @tjdevries  @justinmk

Reply:
> I have no idea how to add test case(s) for the project and run them. Is there any documentation to help to write test cases? If no, I will study it.

See `test/functional/example_spec.lua` for example. You can [run the test](https://github.com/neovim/neovim/blob/master/test/README.md#running-tests) like this:

    TEST_FILE=test/functional/example_spec.lua make functionaltest

More docs at https://github.com/neovim/neovim/blob/master/test/README.md